### DISCUSSION TITLE
WebVisualizer extension not available after pip install?

### DISCUSSION THREAD

**Discussion by justdigi** [isl-org/Open3D#6304] [Q&A] [answered]
Hi, I'm trying to see 3D models inside of Jupyter lab, but I get an error that suggests I'm not quite set up correctly. Can anyone help?

I created a virtualenv for Python 3.10 and used pip to install `open3d` (0.17.0) and `jupyterlab`.

But when I tried the sample code in a notebook, 
```python
import open3d as o3d
from open3d.web_visualizer import draw

cube_red = o3d.geometry.TriangleMesh.create_box(1, 2, 4)
cube_red.compute_vertex_normals()
cube_red.paint_uniform_color((1.0, 0.0, 0.0))
draw(cube_red)
```

I get the following error:
```
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
Cell In[3], line 1
----> 1 from open3d.web_visualizer import draw
      3 cube_red = o3d.geometry.TriangleMesh.create_box(1, 2, 4)
      4 cube_red.compute_vertex_normals()

File ~/.virtualenvs/venv3.10/lib/python3.10/site-packages/open3d/web_visualizer.py:19
     17 from open3d._build_config import _build_config
     18 if not _build_config["BUILD_JUPYTER_EXTENSION"]:
---> 19     raise RuntimeError(
     20         "Open3D WebVisualizer Jupyter extension is not available. To use "
     21         "WebVisualizer, build Open3D with -DBUILD_JUPYTER_EXTENSION=ON.")
     24 @ipywidgets.register
     25 class WebVisualizer(ipywidgets.DOMWidget):
     26     """Open3D Web Visualizer based on WebRTC."""

RuntimeError: Open3D WebVisualizer Jupyter extension is not available. To use WebVisualizer, build Open3D with -DBUILD_JUPYTER_EXTENSION=ON.
```

Thanks for any ideas!

Chris

**Comment by ssheorey** [comment, marked as answer]
Hi @justdigi are you on macOS? Jupyter webvisualizer (server) is not supported on macOS. If not, what is your system environment?

See here for supported platforms: http://www.open3d.org/docs/latest/tutorial/visualization/web_visualizer.html

**Reply by justdigi** [reply]
You're right, I was trying to use MacOS, and I didn't read the **Web Visualizer and Jupyter** correctly. I saw the bottom line in the compatibility table that had a "Yes" MacOS and thought I was good. 

I'd like to suggest that you do a little copy editing of that section and use headings that clearly match each row in the table. Some details on what it means for there to be a Jupyter "client" for MacOS but not a server would be appreciated. The whole client/server distinction in viz/windowing tools like X11 has historically been confusing to many. So, it often deserves more text than you think to help out dummies like me!

Thanks for the follow-up,

Chris

**Reply by ssheorey** [reply]
Thanks for the suggestions. We would appreciate a PR to clarify the documentation. Also, the error message can be made clearer as well.

**Comment by stockdam** [comment]
Visualization is working on my Mac M1 with JupyterLab. It wouldn't work initially and I had to install one of the following (sorry I didn't take notes):

trame
trame-vuetify
trame-vtk
ipywidgets
ipyvtklink

Note that the Visualization window doesn't close properly and you'll have to manually close it for the code to move onto the next cell but I will try the suggestions in the link provided by ssheorey above.
